src/lib/jsonql/promises.js

```diff
@@ -49,7 +60,11 @@ function jsonQL(promises) {
     getPromise({ id, ...others } = {}) {
       const filteredPromises = api.getPromises(others);
       if (id) {
-        return filteredPromises.find((p) => equalsIgnoreCase(p.id, id));
+        const foundPromise = filteredPromises.find((p) =>
+          equalsIgnoreCase(p.id, id)
+        );
+        foundPromise.location = formatLocation(foundPromise.location);
+        return foundPromise;
```

Comment:
Why not format the location when fetching data from gSheet (somewhere [here](https://github.com/CodeForAfrica/PromiseTracker/blob/9b0e0a70a776a4ad9397983e8a738d74e6ead8dc/src/lib/gsheets/index.js#L283-L311) if I'm not mistaken) so that the results are cached?